kge/model/reciprocal_relations_model.py

```diff
@@ -26,10 +27,10 @@ def __init__(
         # Using a dataset with twice the number of relations to initialize base model
         alt_dataset = dataset.shallow_copy()
         alt_dataset._num_relations = dataset.num_relations() * 2
-        reciprocal_relation_ids = [
-            rel_id + "_reciprocal" for rel_id in alt_dataset.relation_ids()
-        ]
-        alt_dataset._meta["relation_ids"].extend(reciprocal_relation_ids)
+        alt_dataset._meta = copy.deepcopy(dataset._meta)
+        alt_dataset._meta["relation_ids"].extend([
+            rel_id + "_reciprocal" for rel_id in dataset.relation_ids()
+        ])
```

Comment:
Good catch! Copying everything is too heavy though (meta data may actually be quite substantial). What we want is shallow copy of the dict and a deep copy of just the relation_ids.

Reply:
Right, that makes sense.